pkg/auth/authenticator/password/htpasswd/htpasswd.go

```diff
@@ -136,7 +139,7 @@ func testPassword(password, hash string) (bool, error) {
 	case strings.HasPrefix(hash, "{SHA}"):
 		// SHA-1, insecure
 		return testSHAPassword(password, hash[5:])
-	case len(hash) == 13:
+	case len(hash) == 13 || strings.HasPrefix(hash, "$"):
```

Comment:
should probably match more judiciously

des: len==13 && !strings.HasPrefix(hash, "$")
md5: $1$
sha256: $5$
sha512: $6$